src/server/services/wms/qgsdxfwriter.cpp

```diff
@@ -35,6 +36,11 @@ namespace QgsWms
 
     // Write output
     QgsRenderer renderer( context );
+
+    //Layer settings need to be kept until QgsDxfExport::writeToFile has finished
+    std::unique_ptr<QgsWmsRestorer> restorer;
```

Comment:
Can you use make_unique? 

```suggestion
    std::unique_ptr<QgsWmsRestorer> restorer = std::make_unique<QgsWmsRestorer>( context );
```